plugins/org.eclipse.n4js.flowgraphs/src/org/eclipse/n4js/flowgraphs/analyses/DirectPathAnalyses.java

```diff
@@ -0,0 +1,226 @@
+/**
+ * Copyright (c) 2017 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.flowgraphs.analyses;
+
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.eclipse.n4js.flowgraphs.ControlFlowType;
+import org.eclipse.n4js.flowgraphs.FGUtils;
+import org.eclipse.n4js.flowgraphs.N4JSFlowAnalyses;
+import org.eclipse.n4js.flowgraphs.model.ComplexNode;
+import org.eclipse.n4js.flowgraphs.model.ControlFlowEdge;
+import org.eclipse.n4js.flowgraphs.model.FlowGraph;
+import org.eclipse.n4js.flowgraphs.model.Node;
+import org.eclipse.n4js.n4JS.ControlFlowElement;
+
+import com.google.common.collect.Lists;
+
+/**
+ * Contains algorithms to reason about single paths between two {@link ControlFlowElement}s.
+ */
+public class DirectPathAnalyses {
+	private final FlowGraph cfg;
+	private final SuccessorPredecessorAnalysis spa;
+
+	/** Constructor */
+	public DirectPathAnalyses(FlowGraph cfg) {
+		this.cfg = cfg;
+		this.spa = new SuccessorPredecessorAnalysis(cfg);
```

Comment:
this creates a `SuccessorPredecessorAnalysis` and containing class `N4JSFlowAnalyses` with crate another instance of `SuccessorPredecessorAnalysis`
Isn't this a redundancy?